tests/lean/Demo/Demo.lean

```diff
@@ -8,6 +8,31 @@ set_option linter.unusedVariables false
 
 namespace demo
 
+/- [demo::CList]
+   Source: 'tests/src/demo.rs', lines 37:0-40:1 -/
+inductive CList (T : Type) where
+| CCons : T → CList T → CList T
+| CNil : CList T
+
+/- Trait declaration: [demo::Counter]
+   Source: 'tests/src/demo.rs', lines 100:0-102:1 -/
+structure Counter (Self : Type) where
+  incr : Self → Result (Usize × Self)
+
+/- [demo::{demo::Counter for usize}::incr]:
+   Source: 'tests/src/demo.rs', lines 105:4-109:5 -/
+def CounterUsize.incr (self : Usize) : Result (Usize × Usize) :=
+  do
+  let self1 ← self + 1#usize
+  ok (self, self1)
+
+/- Trait implementation: [demo::{demo::Counter for usize}]
+   Source: 'tests/src/demo.rs', lines 104:0-110:1 -/
+@[reducible]
+def CounterUsize : Counter Usize := {
+  incr := CounterUsize.incr
+}
+
```

Comment:
This is a problematic reordering: no definition in the `Demo.lean` file should move.
As a side note: when I give demos during presentations I show the Rust file and the Lean file side by side. If the order was not the same, the tool would look a bit broken.